contracts/sponsorship/SponsorshipPaymaster.sol

```diff
@@ -233,11 +233,10 @@ contract SponsorshipPaymaster is
             bytes calldata signature
         )
     {
-        (paymasterId, validUntil, validAfter, priceMarkup) = abi.decode(
-            paymasterAndData[VALID_PND_OFFSET:SIGNATURE_OFFSET],
-            (address, uint48, uint48, uint32)
-        );
-
+        paymasterId = address(bytes20(paymasterAndData[VALID_PND_OFFSET:VALID_PND_OFFSET+20]));
+        validUntil = uint48(bytes6(paymasterAndData[40:46]));
+        validAfter = uint48(bytes6(paymasterAndData[46:52]));
+        priceMarkup = uint32(bytes4(paymasterAndData[52:56]));
```

Comment:
should these be in terms of `VALID_PND_OFFSET` as mentioned above?

Reply:
yeah I was thinking what would be the best?
VALID_PND_OFFSET, VALID_PND_OFFSET+20
VALID_PND_OFFSET+20, VALID_PND_OFFSET+20+6

and so on

then I don't really need SIGNATURE_OFFSET also

or I can remove offsets all together and stick to this numbers only 
wdyt? @ankurdubey521

Reply:
I think the former is fine

Reply:
done